src/ucp/wireup/wireup_ep.c

```diff
@@ -620,6 +620,38 @@ unsigned ucp_wireup_ep_pending_extract(ucp_wireup_ep_t *wireup_ep,
     return count;
 }
 
+void ucp_wireup_eps_pending_extract(ucp_ep_t *ucp_ep,
+                                    ucp_wireup_ep_extract_pending_cb extract_cb,
+                                    ucs_queue_head_t *queue)
+{
+    ucp_lane_index_t lane_idx;
+    ucp_wireup_ep_t *wireup_ep;
+    uct_ep_h uct_ep;
+    int pending_count;
+
+    UCP_WORKER_THREAD_CS_CHECK_IS_BLOCKED(ucp_ep->worker);
+    ucs_assert(extract_cb != NULL);
+    ucs_queue_head_init(queue);
+
+    if (ucp_ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) {
+        return;
+    }
+
+    for (lane_idx = 0; lane_idx < ucp_ep_num_lanes(ucp_ep); ++lane_idx) {
+        uct_ep = ucp_ep_get_lane(ucp_ep, lane_idx);
+        /* When creating EP with remote worker address
+         * EP is using transport lanes only, with no CM lane. */
+        if ((uct_ep == NULL) || (ucp_wireup_ep(uct_ep) == NULL)) {
+            continue;
+        }
+
+        wireup_ep     = ucp_wireup_ep(uct_ep);
+        pending_count = ucp_wireup_ep_pending_extract(wireup_ep, queue);
+        ucp_worker_flush_ops_count_add(ucp_ep->worker, -pending_count);
```

Comment:
we could sum up pending_count and call ucp_worker_flush_ops_count_add once after the loop

Reply:
Good point